### PR TITLE
Optimize workflow of parallel

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
/kind improvement
/cc @czs007

### CODE REVIEW

**Comment by DragonDriver** [comment]
issue: 15000

File: internal/util/funcutil/parallel.go [hunk 1]
```diff
@@ -98,6 +109,7 @@ func ProcessFuncParallel(total, maxParallel int, f func(idx int) error, fname st
 		select {
 		case err := <-errc:
 			close(quit)
+			wg.Wait()
```
Comment (czs007) [inline internal/util/funcutil/parallel.go:112]: should call wg.Wait when count == routineNum

Comment (DragonDriver) [reply internal/util/funcutil/parallel.go:112]: OK

**Pull request merged**